src/KubeOps/Operator/Entities/Extensions/KubernetesObjectExtensions.cs

```diff
@@ -0,0 +1,63 @@
+﻿using k8s;
+using k8s.Models;
+using Org.BouncyCastle.Asn1.Tsp;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace KubeOps.Operator.Entities.Extensions
+{
+    public static class KubernetesObjectExtensions
+    {
+        public static TResource WithOwnerReference<TResource>(this TResource resource, IKubernetesObject<V1ObjectMeta> owner)
+            where TResource : IKubernetesObject<V1ObjectMeta>
+        {
+            resource.Metadata.EnsureOwnerReferences().Add(owner.MakeOwnerReference());
+            return resource;
+        }
+
+        public static IList<V1OwnerReference> EnsureOwnerReferences(this V1ObjectMeta meta)
+        {
+            if (meta.OwnerReferences == null) meta.OwnerReferences = new List<V1OwnerReference>();
+            return meta.OwnerReferences;
+        }
+
+        public static V1OwnerReference MakeOwnerReference(this IKubernetesObject<V1ObjectMeta> kubernetesObject)
+            => new V1OwnerReference(kubernetesObject.ApiVersion, kubernetesObject.Kind, kubernetesObject.Metadata.Name, kubernetesObject.Metadata.Uid);
+
+        public static V1ObjectReference MakeObjectReference(this IKubernetesObject<V1ObjectMeta> kubernetesObject)
+            => new V1ObjectReference()
+            {
+                ApiVersion = kubernetesObject.ApiVersion,
+                Kind = kubernetesObject.Kind,
+                Name = kubernetesObject.Metadata.Name,
+                NamespaceProperty = kubernetesObject.Metadata.NamespaceProperty,
+                ResourceVersion = kubernetesObject.Metadata.ResourceVersion,
+                Uid = kubernetesObject.Metadata.Uid
+            };
+
+        /// <summary>
+        /// Create an event for a kubernetesObject that can be applied to the cluster using the Create on <see cref="Client.IKubernetesClient" />
+        /// </summary>
+        /// <param name="kubernetesObject">The object to create an event for</param>
+        /// <param name="reason">A short reason tag, like Created, Updated, Reconciled</param>
+        /// <param name="message">A human readable message to go with the event</param>
+        /// <param name="type">A string corresponding to the event type, Normal or Warning</param>
+        /// <param name="component">The component generating the event, it should be the name of the operator</param>
+        /// <returns></returns>
+        public static V1Event Event(this IKubernetesObject<V1ObjectMeta> kubernetesObject, string reason, string message, string type = "Normal", string? component = null) =>
```

Comment:
The component (if this should be the operator name) could be ignored and fetched from OperatorSettings right? or should this be more precice? like "operatorname.entitycontroller.type" ?

Reply:
http://localhost:8001/api/v1/namespaces/default/events/

This shows a list of events, of which this is one from my cluster:

```json
{
"metadata": {
"name": "voyager-main-ingress.162c2d50d3052471",
"namespace": "default",
"selfLink": "/api/v1/namespaces/default/events/voyager-main-ingress.162c2d50d3052471",
"uid": "0c787daa-07ad-4c6e-9a0a-d266c527f1c6",
"resourceVersion": "6296576",
"creationTimestamp": "2020-08-30T00:59:40Z"
},
"involvedObject": {
"kind": "Service",
"namespace": "default",
"name": "voyager-main-ingress",
"uid": "5c142ba3-f21a-490f-a158-6cee91f56256",
"apiVersion": "v1",
"resourceVersion": "6296381"
},
"reason": "EnsuredLoadBalancer",
"message": "Ensured load balancer",
"source": {
"component": "service-controller"
},
"firstTimestamp": "2020-08-17T22:03:12Z",
"lastTimestamp": "2020-08-30T01:00:11Z",
"count": 703,
"type": "Normal",
"eventTime": null,
"reportingComponent": "",
"reportingInstance": ""
}
```

The component is a string here which should be the operator name. In this case it is `service-controller`.